src/Sulu/Bundle/PreviewBundle/Resources/js/views/Preview/Preview.js

```diff
@@ -0,0 +1,265 @@
+                return;
+            }
+
+            this.updatePreview(toJS(resourceStore.data));
+        });
+
+        this.started = true;
+    };
+
+    updatePreview = debounce((data) => {
+        const {
+            router: {
+                attributes: {
+                    locale,
+                    webspace,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('update', {
+            locale: locale,
+            webspace: webspace,
+            token: this.token,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.post(route, {data: data}).then((response) => {
+            const document = this.getPreviewDocument();
+            if (!document) {
+                return;
+            }
+
+            document.open();
+            document.write(response.content);
+            document.close();
+        });
+    }, previewConfigStore.debounceDelay);
+
+    updateContext = debounce((type) => {
+        const {
+            router: {
+                attributes: {
+                    webspace,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('update-context', {
+            webspace: webspace,
+            token: this.token,
+        });
+        if (!route) {
+            return;
+        }
+
+        Requester.post(route, {context: {template: type}}).then((response) => {
+            const document = this.getPreviewDocument();
+            if (!document) {
+                return;
+            }
+
+            document.open();
+            document.write(response.content);
+            document.close();
+        });
+    }, previewConfigStore.debounceDelay);
+
+    componentWillUnmount() {
+        if (this.typeDisposer) {
+            this.typeDisposer();
+        }
+
+        if (this.dataDisposer) {
+            this.dataDisposer();
+        }
+
+        if (!this.started) {
+            return;
+        }
+
+        const route = previewConfigStore.generateRoute('stop', {token: this.token});
+        if (!route) {
+            return;
+        }
+
+        Requester.get(route);
+    }
+
+    getPreviewDocument = (): ?Document => {
+        // eslint-disable-next-line
+        const iframe = ReactDOM.findDOMNode(this.iframeRef);
+        if (!(iframe instanceof HTMLIFrameElement)) {
+            return;
+        }
+
+        return iframe.contentDocument;
+    };
+
+    @action setToken = (token: string) => {
+        this.token = token;
+    };
+
+    @action setIframe = (iframeRef: ?Object) => {
+        this.iframeRef = iframeRef;
+    };
+
+    handleToggleSidebarClick = () => {
+        if (sidebarStore.size === 'medium') {
+            return sidebarStore.setSize('large');
+        }
+
+        sidebarStore.setSize('medium');
+    };
+
+    handleRefreshClick = () => {
+        const document = this.getPreviewDocument();
```

Comment:
Couldn't that be cached, instead of being read from the DOM every time?

Reply:
rerendering will break the reference then